### ISSUE TITLE
Dark Souls II: Scholar of the First Sin [BLUS31528]: PS3-native framelimit not honored

### ISSUE THREAD

**Issue by Shoegzer** [RPCS3/rpcs3#16133]
### Quick summary

RPCS3 does not honor setting the framelimit to "PS3 Native" as the title incorrectly runs at 60fps when the settings is on.  This yields inaccuracies in the game experience.

### Details

First, some background. The original PS3 release of this title runs mostly between 20-40fps, as can be seen [here](https://www.youtube.com/watch?v=jQkwnReFURw).  However, when the Scholar of the First Sin edition was released, the version for modern systems got a bump to 60fps while the PS3 version apparently remained at its original variable framerate.  This can be gleaned from the following statement [here](https://www.ign.com/articles/2014/12/29/dark-souls-2-scholar-of-the-first-sin-director-reveals-new-details):

> The higher specs of new-gen consoles and DirectX 11 PCs introduces better graphics, framerate and texture resolution, along with improvements to sound, lighting, shadows, cloth and other effects into the remastered version.

Importantly, players of the next-gen version began to notice weapons degrading faster than they did on the original game.  From Software eventually admitted that this was a bug due to the fact that weapon degradation was _tied to the game's framerate_ and eventually patched it, as described [here](https://www.eurogamer.net/dark-souls-2-patch-will-fix-weapon-degradation-bug).  However, it was never patched for older systems including the PS3, presumably since degradation was occuring as expected with it's slower framerate.

For this reason, it is critical to run this game with the PS3-native framerate to ensure an experience that is accurate and true to the original.  However, as noted above, RPCS3 does not honor the setting for some reason - the game runs at 60fps whether set to "Auto" or "PS3 native".  Another workaround is to set the Framelimit to 30fps but that's not accurate either, as weapon degradation (and apparently other game elements such as roll "iframes" and parrying) is based on the original game's variable framerate.

### Attach a log file

N/A

### Attach capture files for visual issues

N/A

### System configuration

* OS: Linux Mint 22 / kernel 6.8
* CPU: Intel i7-10870
* GPU: Nvidia GTX 3070 / driver 550.107

**Comment by FlexBy420** [comment]
No, Dark Souls 2 targets 60 FPS, the fact that it doesn't reach those FPS on PS3 doesn't change that fact.

**Comment by Shoegzer** [comment]
True, except that's the point of RPCS3's "PS3 native" framelimit option.

**Comment by Shoegzer** [comment]
Is there really a need to be caustic about this?

Anyway, I thought that was the point of the "Auto" setting - to auto-lock the game to its targeted framerate, whereas the "PS3 native" setting would attempt to match the speed of the actual PS3 based on PPU/SPU clock cycles (otherwise why have the feature)?  I'm not sure if it's documented anywhere exactly how the feature works, so I could be wrong, but that's my best guess.

Playing the game as it exists introduces noticeable inaccuracies that I observed myself.  And no, you obviously wouldn't do all those manual steps to workaround the issue, don't be silly.

**Comment by Shoegzer** [comment]
@schm1dtxbox Interesting, that makes sense. Given what you've said there really isn't a proper way to fix the problem, even with proper vblank/flip behavior, other than to lock the title to 30fps as I mentioned.  I suppose @elad335 could confirm this if he has time, since I think he wrote that feature, but I assume you're right.

Maybe someday faster host system hardware will allow such a feature (perhaps a cycle-accurate SPU would help if it could ever be done realistically), and the game will play more like it's supposed to.  I'm sure there are other titles that lock game elements to the framerate and would benefit from such a thing.

Anyway, I'll close this issue under the assumption there's not much that can be done about it, to avoid cluttering up the issue log, but if you find anything that suggests otherwise just let me know and I'll reopen.  Thanks for your thoughts!

**Comment by kd-11** [comment]
Let's first get evidence of the 30fps lock on PS3. Fromsoft usually does not do real 30fps lock since DeS. Even DS3 was infamously just running unlocked.
If anyone has access to retail PS3 hardware and can test it it would be useful. Especially when looking at the sky.
There is also another issue of unlocked fps + vsync which can give false appearance of 30fps lock on console but 60fps on PC.

**Comment by Shoegzer** [comment]
@kd-11 unless I'm missing something, doesn't the DF video in my OP (repost [here](https://www.youtube.com/watch?v=jQkwnReFURw)) show that there is no such lock on PS3 hardware?  You'll notice in most areas that the PS3 runs the game between 25-30fps, though at times it does briefly reach 40 fps or above.

**Comment by kd-11** [comment]
@Shoegzer If it goes over 30fps then it is unlocked with 60fps vsync, in which case I don't see any issue here. Obviously a modern PC is more powerful than a PS3 so the fps would be higher. Cycle-accurate emulation is impossible due to the high clock speed of the PS3 processor (3.2GHz).

**Comment by Shoegzer** [comment]
Agreed, and that's why I closed the issue but I was hoping @elad335 might have some ideas when he re-opened it.  If not, I or someone else can close it again.  

And yeah, given the high requirements to emulate even a 16-bit 65C816 CPU at cycle accuracy, I can't imagine emulating a PS3 at that level will even be possible in our lifetimes.  It's unfortunate as it seems this title will never really run the way it was intended, at least under RPCS3, given my experiences above.  The only workaround then would be to lock the game to 30fps to approximate the experience, as I mentioned, unless you know of a better way.

**Comment by kd-11** [comment]
Well, in that case the title is a bit misleading. PS3-native frame limit doesn't mean we lock to some value matching PS3, it just means that it uses timings similar to a PS3 to trigger vblank.
There's no real solution to the higher fps on PC other than simply applying a FPS limit. I'll leave it open since I'm not the assigned developer.

**Comment by digant73** [comment]
@Shoegzer is that still an issue?

**Comment by Shoegzer** [comment]
I'm not aware of any work done on this over the past year, so I would leave it open. That said, from what @kd-11 stated above I'm not sure whether it's even possible.  I will leave this up to @elad335 to close if necessary, though I'm sure he had a reason to re-open it.
